### ISSUE TITLE
Bitbucket Push not triggering pipeline build

### ISSUE THREAD

**Issue by jbmolle** [jenkinsci/bitbucket-branch-source-plugin#204]
### Your checklist for this issue

- [x] Jenkins version
Docker image 1.17.0  · Core 2.181  · 449ec0e  · 7th June 2019 10:15 PM
- [x] Plugin version
2.4.4
- [x] Bitbucket cloud

- [ ] Bitbucket server and version

### Description

Hi,

I'm using BlueOcean / Bitbucket plugin and my repositories are on Bitbucket Cloud.
I've set up a webhook in Bitbucket to https://{JENKINS_URL}/bitbucket-scmsource-hook/notify
When I do a push on the repo, the webhook is correctly triggered. I can see it in Bitbucket and I receive a 200 response.

On Jenkins, I get the following:
- When I enable plugin logs: X-Bitbucket-Type header / server_url request parameter not found. Bitbucket Cloud webhook incoming. 
So I guess this is working correctly because I'm using Bitbucket Cloud.
- On /var/jenkins_home/logs/jenkins.branch.MultiBranchProject.log, I get "Received com.cloudbees.jenkins.plugins.bitbucket.hooks.PushHookProcessor$1 UPDATED event..." and then "Finished processing com.cloudbees.jenkins.plugins.bitbucket.hooks.PushHookProcessor$1 UPDATED event... Matched 0."

So it seems that the plugin doesn't find a corresponding Bitbucket repository when receiving the hook event. It's strange because I logged in Bitbucket Cloud with BlueOcean and it can list all my repositories. I select the repository with the dropdown so I can't make any mistake here.

Do you have any idea if it's a bug or if I'm doing something wrong?

Thanks,
JB

**Comment by jbmolle** [comment]
Hi everyone,
So I think I've found where the bug is coming from.

When I create my pipeline with BlueOcean, I can choose the team and it discovers all the repositories. If I click on "Configure" from BlueOcean Dashboard, I can see "Branch Sources" under "General" tab. Bitbucket is auto configured and the Owner is in the form {xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx} which is the owner uuid on Bitbucket.

In BitbucketCloudRepository.java, the getOwnerName() and getRepositoryName() are derived from full_name send by Bitbucket post hook request body so in the form my_username/my_repository_name.

Therefore I was getting Matched0 because the plugin cannot find {xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx}/my_repository_name.

Now I went back to the "Branch Sources" and I changed Owner from {xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx} to my_username. The plugin finds all repository and the hook from Bitbucket generates a Matched 1 and triggers the pipeline.

So in the end, the bug lies in the fact that BlueOcean is saving owner uuid and the BitbucketCloudRepository is getting the OwnerName from hook full_name. It should parse instead "owner" "uuid". Or BlueOcean should save "owner" "username" instead of "owner" "uuid"

Let me know what you think.

Thanks,
JB

**Comment by caiohasouza** [comment]
I have the same problem.

**Comment by tuanalumi** [comment]
Any update on this?

**Comment by casz** [comment]
Anyone is welcome to pick up the issue and create a PR

**Comment by qq7987590** [comment]
same problem.

**Comment by JamesMc86** [comment]
Hit the same issue. This appears to be an intentional change in BlueOcean as they say the username is going away?
Found this issue which is the same problem: https://issues.jenkins-ci.org/browse/JENKINS-58019

**Comment by SubrataM3** [comment]
> Anyone is welcome to pick up the issue and create a PR

This issue seems to be fixed with the PR #182  but the PR is not merged. 
@casz  Any update on this? I have the same issue.

Thanks.

**Comment by casz** [comment]
Same as previous
> Anyone is welcome to pick up the issue and create a PR

I am only a maintainer, I don't actively use bitbucket. If someone wants to actively maintain the Bitbucket integration I'd gladly accept a maintainer who actively uses Bitbucket. I'll also gladly review any contributions.

**Comment by casz** [comment]
Also as noted in this comment #182 is not needed: https://github.com/jenkinsci/bitbucket-branch-source-plugin/pull/182#issuecomment-501768296

**Comment by SubrataM3** [comment]
Okay, I see that the PR will not fix the issue. the plugin is really expecting server_url query string (which i think is required only for bitbucket server). If someone have a workaround kindly post .

**Comment by sneko** [comment]
Same... sometimes manually updating ownerId by nickname works, sometimes not... driving me crazy 😄

**Comment by medicce** [comment]
We're having issues with this as well.
- I can confirm in Bitbucket Cloud that the webhook is firing and being accepted (200), In Jenkins, 

- I confirm receipt in the access logs. 

- com.cloudbees.jenkins.plugins.bitbucket.hooks.BitbucketSCMSourcePushHookReceiver shows that the webhook was incoming. 

- com.cloudbees.jenkins.plugins.bitbucket.hooks.PushHookProcessor$1 confirms that it matched the job and the branch.

- As mentioned above, $JENKINS_HOME/logs/jenkins.branch.MultiBranchProject.log was showing "Matched 0". When I changed the UUID to the team name, it started showing "Matched 1", but still wasn't triggering.

- In $JENKINS_HOME/logs/jenkins.branch.OrganizationFolder.log, it was showing "Matched 0" on every webhook received, though I'm unsure what this means.

I'm using 
Bitbucket Cloud
Jenkins 2.195
Bitbucket Branch Source Plugin 2.5

**Comment by sneko** [comment]
No stale

**Comment by sammosov** [comment]
Guys, I`ve managed to make it work with /bitbucket-scmsource-hook/notify endpoint AND --- enable "Manage Hooks" in Configure System --- > Bitbucket Endpoints --> Manage hooks and set the credential with admin rights in BitBucked cloud. Plugin version is 2.6.0. According to this https://support.cloudbees.com/hc/en-us/articles/115000051132-How-to-Trigger-Multibranch-Jobs-from-Bitbucket-Cloud-  There is no need to do this.  Please fix the Plugin behavior if possible.

**Comment by unixhat** [comment]
same happen here.. got same problem.

**Comment by unixhat** [comment]
> Guys, I`ve managed to make it work with /bitbucket-scmsource-hook/notify endpoint AND --- enable "Manage Hooks" in Configure System --- > Bitbucket Endpoints --> Manage hooks and set the credential with admin rights in BitBucket cloud. Plugin version is 2.6.0. According to this https://support.cloudbees.com/hc/en-us/articles/115000051132-How-to-Trigger-Multibranch-Jobs-from-Bitbucket-Cloud- There is no need to do this. Please fix the Plugin behavior if possible. So at least you can add the statement into the article that **setting Manage Hooks with BitBucket Credential is the requirement.**

It's working good for my side for the first time, after that i trying to push again and got same problem. Trying to switch to Blueocean crenden and switch back to bibucket cloud creden same happen.

**Comment by KangoV** [comment]
We're having this exact problem

**Comment by casz** [comment]
https://plugins.jenkins.io/cloudbees-bitbucket-branch-source/
![image](https://user-images.githubusercontent.com/1661688/75810167-5ae44200-5d8a-11ea-9973-752982942dce.png)

**Comment by tgolden-andplus** [comment]
Not stale. Plugin flat-out doesn't work anymore.

**Comment by hmcmanus** [comment]
I had a good hard look at this one and I was able to get it running after some time digging into the code. I have it working for freestyle job types and multi-branch pipeline job types. Still can't get it working for team projects. 

The fix for me was to set a value in the "Override Repository URL or Path" text box. Found it from this part of the code:

https://github.com/jenkinsci/bitbucket-plugin/blob/6b9ff82226167413e1f19f80bd8ee536b9537cc6/src/main/java/com/cloudbees/jenkins/plugins/BitbucketJobProbe.java#L107

After putting the repository URL in there it triggers the job.
